content/manuals/build/cache/garbage-collection.md

```diff
@@ -8,89 +8,266 @@ aliases:
+
+Each GC policy is evaluated in sequence, starting with the most specific
+criteria, and proceeds to broader rules if previous policies do not resolve the
+issue. This lets BuildKit prioritize cache entries, preserving the most
+valuable cache while ensuring the system maintains performance and
+availability.
+
+The default GC policies are (approximately):
+
+1. Remove cache that can be easily regenerated, such as build contexts from
+   local directories or remote Git repositories, and cache mounts, if hasn't
+   been used for more than 48 hours.
+2. Remove cache that hasn't been used in a build for more than 60 days.
+3. Remove unshared cache that exceeds the build cache size limit. Unshared
+   cache records refers to layer blobs that are not used by other resources
+   (typically, as image layers).
+4. Remove any build cache that exceeds the build cache size limit.
+
+The precise algorithm and the means of configuring the policies differ slightly
+depending on what kind of builder you're using. Refer to
+[Configuration](#configuration) for more details.
 
 ## Configuration
 
-Depending on the [driver](../builders/drivers/_index.md) used by your builder instance,
-the garbage collection will use a different configuration file.
+> [!NOTE]
+> If you're satisfied with the default garbage collection behavior and don't
+> need to fine-tune its settings, you can skip this section. Default
+> configurations work well for most use cases and require no additional setup.
+
+Depending on the type of [build driver](../builders/drivers/_index.md) you use,
+you will use different configuration files to change the builder's GC settings:
+
+- If you use the default builder for Docker Engine (the `docker` driver), use
+  the [Docker daemon configuration file](#docker-daemon-configuration-file).
+- If you use a custom builder, use a [BuildKit configuration file](#buildkit-configuration-file).
+
+### Docker daemon configuration file
+
+If you're using the default [`docker` driver](../builders/drivers/docker.md),
+GC is configured in the [`daemon.json` configuration file](/reference/cli/dockerd.md#daemon-configuration-file),
+or if you use Docker Desktop, in [**Settings > Docker Engine**](/manuals/desktop/settings-and-maintenance/settings.md).
+
+The following snippet shows the default builder configuration for the `docker`
+driver for Docker Desktop users:
+
+```json
+{
+  "builder": {
+    "gc": {
+      "defaultKeepStorage": "20GB",
+      "enabled": true
+    }
+  }
+}
+```
+
+The `defaultKeepStorage` option configures the size limit of the build cache,
+which influences the GC policies. The default policies for the `docker` driver
+work as follows:
 
-If you're using the [`docker` driver](../builders/drivers/docker.md), garbage collection
-can be configured in the [Docker Daemon configuration](/reference/cli/dockerd.md#daemon-configuration-file).
-file:
+1. Remove ephemeral, unused build cache older than 48 hours if it exceeds 10%
```

Comment:
🙃 

https://github.com/moby/moby/blob/d60e55d613a35a11d59310a1ec16eb06f2133eb1/builder/builder-next/worker/gc.go#L12-L15

```suggestion
1. Remove ephemeral, unused build cache older than 48 hours if it exceeds 13.8%
```

Reply:
We should align moby with buildkit, doesn't make sense to have a different default GC policy. Moby could probably just call https://github.com/moby/buildkit/blob/f2092259de4a7fbdb05c84b1c8791c9a3ca578c4/cmd/buildkitd/config/gcpolicy.go#L71 and remove https://github.com/moby/moby/blob/d60e55d613a35a11d59310a1ec16eb06f2133eb1/builder/builder-next/worker/gc.go#L18

cc @thaJeztah @vvoland @tonistiigi

Reply:
```go
// tempCachePercent represents the percentage ratio of the cache size in bytes to temporarily keep for a short period of time (couple of days)
// over the total cache size in bytes. Because there is no perfect value, a mathematically pleasing one was chosen.
// The value is approximately 13.8
const tempCachePercent = math.E * math.Pi * math.Phi
```
> https://github.com/moby/moby/blob/d60e55d613a35a11d59310a1ec16eb06f2133eb1/builder/builder-next/worker/gc.go#L12-L15

lol